swagger_server/test_local/test_ciupload_controller.py

```diff
@@ -0,0 +1,313 @@
+##############################################################################
+#                                                                            #
+#   Collection Instruments Upload                                            #
+#   Date:    11 May 2017                                                     #
+#   Author:  Gareth Bult                                                     #
+#   License: MIT                                                             #
+#   Copyright (c) 2017 Crown Copyright (Office for National Statistics)      #
+#                                                                            #
+##############################################################################
+from . import BaseTestCase
+from werkzeug.datastructures import FileStorage
+from six import BytesIO
+from ..controllers_local.collectioninstrument import CollectionInstrument
+from uuid import uuid4
+
+DEFAULT_SURVEY = "3decb89c-c5f5-41b8-9e74-5033395d247e"
+
+
+class TestCiuploadController(BaseTestCase):
+    """ CiuploadController integration test stubs """
+
+    def setUp(self):
+        """
+        The collection_instrument variable represents the external class we're about to exercise ...
+        """
+        super()
+        self.collection_instrument = CollectionInstrument()
+
+    def test_01_define_batch(self):
+        """
+        Make sure we can define a batch
+        """
+        code, msg = self.collection_instrument.define_batch(str(uuid4()), 100)
+        self.assertTrue(code == 200, msg)
+        try:
+            code, msg = self.collection_instrument.define_batch(0, 100)
+            self.assertTrue(code == 200, msg)
+        except:
+            pass
+
+    def test_03_upload(self):
+        """
+        Try a valid upload, and also an upload to an undefined batch
+        """
+        with open('scripts/upload.txt', 'rb') as io:
+            fileobject = FileStorage(stream=io, filename='dummy.txt', name='myname')
+            code, msg = self.collection_instrument.upload(str(uuid4()), fileobject, 'dummy.txt')
+            self.assertTrue(code == 200, msg)
+
+        with open('scripts/upload.txt', 'rb') as io:
+            fileobject = FileStorage(stream=io, filename='dummy.txt', name='myname')
+            code, msg = self.collection_instrument.upload(0, fileobject, 'dummy.txt')
+            self.assertTrue(code == 400, msg)
+
+    def test_04_csv(self):
```

Comment:
Numbering ensures the tests are run in order rather than randomly .. when developing tests and watching output it is **very** helpful, at the very lest to me. Test functions are all commented, why do we need more descriptive function names?